datafusion/optimizer/src/single_distinct_to_groupby.rs

```diff
@@ -64,63 +67,55 @@ impl SingleDistinctToGroupBy {
 }
 
 /// Check whether all aggregate exprs are distinct on a single field.
-fn is_single_distinct_agg(plan: &LogicalPlan) -> Result<bool> {
-    match plan {
-        LogicalPlan::Aggregate(Aggregate { aggr_expr, .. }) => {
-            let mut fields_set = HashSet::new();
-            let mut aggregate_count = 0;
-            for expr in aggr_expr {
-                if let Expr::AggregateFunction(AggregateFunction {
-                    func_def: AggregateFunctionDefinition::BuiltIn(fun),
-                    distinct,
-                    args,
-                    filter,
-                    order_by,
-                    null_treatment: _,
-                }) = expr
-                {
-                    if filter.is_some() || order_by.is_some() {
-                        return Ok(false);
-                    }
-                    aggregate_count += 1;
-                    if *distinct {
-                        for e in args {
-                            fields_set.insert(e.canonical_name());
-                        }
-                    } else if !matches!(fun, Sum | Min | Max) {
-                        return Ok(false);
-                    }
-                } else if let Expr::AggregateFunction(AggregateFunction {
-                    func_def: AggregateFunctionDefinition::UDF(fun),
-                    distinct,
-                    args,
-                    filter,
-                    order_by,
-                    null_treatment: _,
-                }) = expr
-                {
-                    if filter.is_some() || order_by.is_some() {
-                        return Ok(false);
-                    }
-                    aggregate_count += 1;
-                    if *distinct {
-                        for e in args {
-                            fields_set.insert(e.canonical_name());
-                        }
-                    } else if fun.name() != "SUM"
-                        && fun.name() != "MIN"
-                        && fun.name() != "MAX"
-                    {
-                        return Ok(false);
-                    }
-                } else {
-                    return Ok(false);
+fn is_single_distinct_agg(aggr_expr: &[Expr]) -> Result<bool> {
+    let mut fields_set = HashSet::new();
+    let mut aggregate_count = 0;
+    for expr in aggr_expr {
+        if let Expr::AggregateFunction(AggregateFunction {
+            func_def: AggregateFunctionDefinition::BuiltIn(fun),
+            distinct,
+            args,
+            filter,
+            order_by,
+            null_treatment: _,
+        }) = expr
+        {
+            if filter.is_some() || order_by.is_some() {
+                return Ok(false);
+            }
+            aggregate_count += 1;
+            if *distinct {
+                for e in args {
+                    fields_set.insert(e.canonical_name());
```

Comment:
I think we can reduce string allocations here:

```suggestion
                    fields_set.insert(e);
```

Reply:
removed in https://github.com/apache/datafusion/pull/10527/commits/0f069b4c68072239b99e3081776ef447f5cfe885